### DISCUSSION TITLE
Test Fail Criteria based on Baseline score

### DISCUSSION THREAD

**Discussion by chdittmer** [stryker-mutator/stryker-net#3123] [Ideas]
I would like to be able to set a fail criteria based on the baseline mutation score. Basically I would like to incentivize our developers to write better unit tests by ensuring any new changes to code has better unit test coverage by ensuring the new tests end up with a better mutation score than the baseline.

Required behavior:
- Set the baseline mutation score as the "break" criteria (Anything lower than baseline will break).
- Do not create a new baseline if the run "breaks" due to the score being worse.
- If mutation score is better continue as usual

This way our CI/CD pipelines will automatically be able stop any new (and potentially untested) code from being merged into a release.

**Comment by rouke-broersma** [comment]
This seems like a good feature, feel free to create a new feature request for it.

**Comment by chdittmer** [comment]
https://github.com/stryker-mutator/stryker-net/issues/3124

Thanks I have done just that
